src/types.rs

```diff
@@ -1709,6 +2450,47 @@ impl TRS {
                 .all(|(r1, r2)| Rule::unify(r1, r2).is_some())
     }
     /// Does one TRS match another?
+    ///
+    /// See [`Pattern Matching`] for more information.
```

Comment:
For clarity and conciseness, please move the link up to `match` in the previous sentence and then delete this sentence.